docs: update KubePersistentVolumeFillingUp with prerequisite for store-gateways (#9939)
grafana/mimir: docs/sources/mimir/manage/mimir-runbooks/_index.md

    - Check the number of in-memory series shown on the `Mimir / Tenants` dashboard for an approximation of the number of series that will be compacted once these blocks are shipped from ingesters.
    - Check the configured `compactor_split_and_merge_shards` for the tenant. A reasonable rule of thumb is 8-10 million series per compactor shard - if the number of series per shard is above this range, increase `compactor_split_and_merge_shards` for the affected tenant(s) accordingly.

##### Persistent Volume Resizing Prerequisite - jsonnet

If persistent volume resizing is required for store-gateways and automatic downscaling is enabled, it must be disabled before proceeding with the resizing process. This step is necessary to prevent any unexpected downscaling by the rollout operator while updating the stateful set for each zone.

> Disable automatic downscaling for store-gateways by setting `store_gateway_automated_downscale_enabled` equal to `false`.

Reviewer: was this in `$._config.store_gateway_automated_downscale_enabled` or just `$.store_gateway_automated_downscale_enabled`? maybe you can have a full jsonnet snippet here like

```
{
  _config+: {
    store_gateway_automated_downscale_enabled: flase
  }
}
```

Author: The first one. You are right. Let me add a snippet . Thank you

Author: Added here [5bbcbcc](https://github.com/grafana/mimir/pull/9939/commits/5bbcbcc7863dc866be5ac4eecfa3fcce5d8bb5be)

    - Check the number of in-memory series shown on the `Mimir / Tenants` dashboard for an approximation of the number of series that will be compacted once these blocks are shipped from ingesters.
    - Check the configured `compactor_split_and_merge_shards` for the tenant. A reasonable rule of thumb is 8-10 million series per compactor shard - if the number of series per shard is above this range, increase `compactor_split_and_merge_shards` for the affected tenant(s) accordingly.

##### Persistent Volume Resizing Prequisite - store-gateways

In case persistent volume resizing is needed for store-gateways, ensure the following steps are taken before proceeding, if automatic downscaling and/or autoscaling, have been enabled:

> - Disable automatic downscaling for store-gateways.

Reviewer 2: Could you please be more specific about what should be done here?

Author: Hello @charleskorn . Thank you for reviewing. I believe the runbook is now more concise and specific. I also considered several comments before drafting it.

- Automatic downscaling is enabled only in the Jsonnet ([Ref](https://github.com/grafana/mimir/blob/main/operations/mimir/store-gateway-automated-downscale.libsonnet)). I couldn't find any reference to it in Helm, which is why I skipped the Helm reference. Please correct me if I’m wrong.
- Autoscaling in both [Jsonnet](https://github.com/grafana/mimir/blob/dc3ddfa730e9a1ddb2f74c34ade885ae7b6d245f/operations/mimir/multi-zone.libsonnet#L22) and [Helm](https://github.com/grafana/mimir/blob/dc3ddfa730e9a1ddb2f74c34ade885ae7b6d245f/operations/helm/charts/mimir-distributed/values.yaml#L2009) is similar. We configure a number based on a variable, which is then divided by 3 to determine the desired number of replicas for each StatefulSet in each zone. Therefore, I skipped it from the prerequisite list.

Let me know if it makes sense to you. Thank you

    - Check the number of in-memory series shown on the `Mimir / Tenants` dashboard for an approximation of the number of series that will be compacted once these blocks are shipped from ingesters.
    - Check the configured `compactor_split_and_merge_shards` for the tenant. A reasonable rule of thumb is 8-10 million series per compactor shard - if the number of series per shard is above this range, increase `compactor_split_and_merge_shards` for the affected tenant(s) accordingly.

> ##### Persistent Volume Resizing Prerequisite - jsonnet

Reviewer 3: I'm not sure about this formatting. Is this information an additional step in troubleshooting the store-gateway? If so, should it be integrated into the bullet points above?

Author: Updated [f8c8f15](https://github.com/grafana/mimir/pull/9939/commits/f8c8f158cf498de578713ea88dc6ddf709798781). I added it as a final bullet of the store-gateway `How to **investigate** and **fix** it:`. Let me know if it looks better now

Reviewer 3: I think this looks great, @NickAnge !

> If persistent volume resizing is required for store-gateways and automatic downscaling is enabled, it must be disabled before proceeding with the resizing process. This step is necessary to prevent any unexpected downscaling by the rollout operator while updating the stateful set for each zone.

Reviewer 3: ```suggestion
If persistent volume resizing is required for store-gateways and automatic downscaling is enabled, you must disable it before proceeding with the resizing process. This step is necessary to prevent any unexpected downscaling by the rollout operator while updating the stateful set for each zone.
```

Reviewer 3: We try to use active voice whenever possible, per https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#write-in-active-voice.

> Disable automatic downscaling for store-gateways by setting `$._config.store_gateway_automated_downscale_enabled` equal to `false`. Example:

Reviewer 3: ```suggestion
To disable automatic downscaling for store-gateways, set `$._config.store_gateway_automated_downscale_enabled = false`. For example:
```

```jsonnet
{
    _config+:
    {
        store_gateway_automated_downscale_enabled: false
    }
}
```

> After the resizing process is finished, revert above change.

Reviewer 3: ```suggestion
After the resizing process finishes, revert this change.
```
